app/src/main/java/com/vero/tasky/agenda/domain/worker/SyncAgendaWorker.kt

```diff
@@ -0,0 +1,44 @@
+package com.vero.tasky.agenda.domain.worker
+
+import android.content.Context
+import androidx.hilt.work.HiltWorker
+import androidx.work.CoroutineWorker
+import androidx.work.WorkerParameters
+import com.vero.tasky.agenda.domain.repository.AgendaRepository
+import dagger.assisted.Assisted
+import dagger.assisted.AssistedInject
+import retrofit2.HttpException
+
+@HiltWorker
+class SyncAgendaWorker @AssistedInject constructor(
+    @Assisted private val context: Context,
+    @Assisted private val workerParameters: WorkerParameters,
+    private val agendaRepository: AgendaRepository
+) : CoroutineWorker(context, workerParameters) {
+
+    override suspend fun doWork(): Result {
```

Comment:
If this worker fails, it will retry
And then retry again
And again
and infinitely retry

What you can do, is have a max retry limit
```
if (runAttemptCount >= 3) {
    return Result.failure()
}
```